packages/transaction-controller/src/TransactionController.ts

```diff
@@ -411,26 +480,25 @@ export class TransactionController extends BaseControllerV1<
       getCurrentNetworkEIP1559Compatibility: () => Promise<boolean>;
       getExternalPendingTransactions?: (
         address: string,
+        chainId?: string,
       ) => NonceTrackerTransaction[];
       getGasFeeEstimates?: () => Promise<GasFeeState>;
       getNetworkState: () => NetworkState;
       getPermittedAccounts: (origin?: string) => Promise<string[]>;
       getSavedGasFees?: (chainId: Hex) => SavedGasFees | undefined;
       getSelectedAddress: () => string;
-      incomingTransactions?: {
-        includeTokenTransfers?: boolean;
-        isEnabled?: () => boolean;
-        queryEntireHistory?: boolean;
-        updateTransactions?: boolean;
-      };
+      incomingTransactions?: IncomingTransactionOptions;
```

Comment:
Could these be calls to the `NetworkController` via the messenger rather than additional callbacks?